### PR TITLE
Yunyang/feature/photo taking

### PR DESCRIPTION
**Pull request by Yunyang-Lu**
photo taking for items working now, click image in edit fragment to take a photo for item

Known issues: 
1.  when add an item, if user first input name, mode, etc. then taking a photo, the information inputted will be lost
2. only 1 image per item rn

### CODE REVIEW

File: code/app/src/main/java/com/example/sigma_blue/activities/PhotoTakingActivity.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            imageBitmap.compress(Bitmap.CompressFormat.PNG, 100, outputStream);
+            byte[] data = outputStream.toByteArray();
+
+            String path = globalContext.getAccount().getUsername() + "/" + UUID.randomUUID() +".png";
+            globalContext.getCurrentItem().setPhotoPath(path);
+            StorageReference firebaseStorageRef = storage.getReference(path);
+
+            //now upload the data(image), expected to take some time
+            UploadTask uploadTask = firebaseStorageRef.putBytes (data);
```
Comment (waridh) [inline code/app/src/main/java/com/example/sigma_blue/activities/PhotoTakingActivity.java:76]: Does this already use threading? If not, would we require threading?

File: code/app/src/main/java/com/example/sigma_blue/activities/PhotoTakingActivity.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent Data) {
+
+        super.onActivityResult(requestCode, resultCode, Data);
+
+        if (requestCode == REQUEST_IMAGE_CAPTURE && resultCode == RESULT_OK){
+            Bundle extras = Data.getExtras();
+            Bitmap imageBitmap = (Bitmap) extras.get("data");
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
```
Comment (waridh) [inline code/app/src/main/java/com/example/sigma_blue/activities/PhotoTakingActivity.java:67]: I think that a function that converts Bitmap to byte[] would be good for code quality.

File: code/app/src/main/java/com/example/sigma_blue/entity/item/Item.java [hunk 3]
```diff
@@ -436,7 +449,8 @@ public int hashCode() {
                     q.getString("MAKE"),
                     q.getString("MODEL"),
                     q.getString("SERIAL"),
-                    Float.parseFloat(q.getString("VALUE"))
+                    Float.parseFloat(q.getString("VALUE")),
+                    q.getString("IMAGE")
```
Comment (waridh) [inline code/app/src/main/java/com/example/sigma_blue/entity/item/Item.java:453]: This is using an old version of the Item class. Please rebase/squash development into this branch, and then make a new account and test it. The latest version of the system no longer uses HashMap<String, String>, and instead uses HashMap<String, Object>. This means that there may be some conflicts when merging this into dev.

File: code/app/src/main/java/com/example/sigma_blue/fragments/DetailsFragment.java [hunk 4]
```diff
@@ -112,6 +123,32 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState)
         tagListAdapter = TagListAdapter.newInstance(currentItem.getTags(), getContext());
         tagListView.setAdapter(tagListAdapter);
 
+        // ITEM IMAGE
+        // trying to get the path of image, and put it on the add item
+        String tempImagePath = globalContext.getCurrentItem().getPhotoPath();
+        // set the image of the item
+        // Create a storage reference from our app
+        if (tempImagePath != null) {
```
Comment (waridh) [inline code/app/src/main/java/com/example/sigma_blue/fragments/DetailsFragment.java:131]: We should have handling for when the photo path is null. The current interaction is showing a question mark where the photo should be.

File: code/app/src/main/java/com/example/sigma_blue/fragments/EditFragment.java [hunk 5]
```diff
@@ -136,6 +151,32 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState)
         SimpleDateFormat sdf = new SimpleDateFormat(getResources().getString(R.string.date_format));
         textDate.setText(sdf.format(currentItem.getDate()));
 
+        //ITEM IMAGE RELATED CHANGES
+        // trying to get the path of image, and put it on the add item
+        String tempImagePath = globalContext.getCurrentItem().getPhotoPath();
```
Comment (waridh) [inline code/app/src/main/java/com/example/sigma_blue/fragments/EditFragment.java:156]: Code here looks like a copy of the details view fragment. Should be factored out to an inherited/static function for code reuse.

**Review by waridh — CHANGES_REQUESTED** [review]
## Compatibility testing

Please merge development into this branch and test to see if it runs with the new changes to the Item class and the way that the app loads from the database.  You may need to make a new account for the new version to work.

## Code quality

Also try to factor out some repeated code. We don't want to let the code base get cluttered.

File: code/app/src/main/java/com/example/sigma_blue/entity/item/Item.java (hunk 3 above)
Comment (Yunyang-Lu) [reply code/app/src/main/java/com/example/sigma_blue/entity/item/Item.java:453]: I see the issues happened here. going to adapt the new item class, and trying to use similar structure in develop branch for multi-photo implementation, hopefully finishing tonight.

**Pull request closed without merging**